vector/src/main/java/im/vector/app/features/analytics/impl/DefaultVectorAnalytics.kt

```diff
@@ -41,12 +42,12 @@ private val IGNORED_OPTIONS: Options? = null
 @Singleton
 class DefaultVectorAnalytics @Inject constructor(
         postHogFactory: PostHogFactory,
-        private val sentryFactory: SentryFactory,
+        private val sentryAnalytics: SentryAnalytics,
         analyticsConfig: AnalyticsConfig,
         private val analyticsStore: AnalyticsStore,
         private val lateInitUserPropertiesFactory: LateInitUserPropertiesFactory,
         @NamedGlobalScope private val globalScope: CoroutineScope
-) : VectorAnalytics {
+) : VectorAnalytics, ErrorTracker by sentryAnalytics {
```

Comment:
Should we replace `ErrorTracker by sentryAnalytics` by actual implementation in order to check for user consent in this class, before sending to sentry? This is maybe not necessary, if Sentry does nothing when it is not enabled, but better to double check.

```kotlin
    override fun trackError(throwable: Throwable) {
        sentryAnalytics
                .takeIf { userConsent == true }
                ?.trackError(throwable)
    }
```